### PR TITLE
[WW-5165] Upgrades Spring to version 5.3.20

### PR DESCRIPTION
**Pull request by lukaszlenart**
Refs [WW-5165](https://issues.apache.org/jira/browse/WW-5165)
Closes #534

### CODE REVIEW

File: plugins/portlet/pom.xml [hunk 1]
```diff
@@ -130,11 +130,7 @@
             <artifactId>spring-test</artifactId>
             <optional>true</optional>
         </dependency>
-        <dependency>
```
Comment (sepe81) [inline plugins/portlet/pom.xml:133]: Maybe we could add some additional infos to the commit to clarify the context

Remove unused spring-webmvc-portlet
Spring 5.x dropped support for spring-webmvc-portlet
https://github.com/spring-projects/spring-framework/wiki/Upgrading-to-Spring-Framework-5.x#dropped-support

File: core/pom.xml [hunk 2]
```diff
@@ -242,6 +242,16 @@
             <artifactId>spring-web</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
```
Comment (sepe81) [inline core/pom.xml:247]: background: Add spring-context for testing with StaticApplicationContext

File: plugins/portlet-mocks/pom.xml [hunk 3]
```diff
@@ -51,6 +51,7 @@
     </dependencies>
 
     <properties>
+        <spring.platformVersion>4.3.30.RELEASE</spring.platformVersion>
```
Comment (sepe81) [inline plugins/portlet-mocks/pom.xml:54]: good idea, to freeze the spring version here

**Review by sepe81 — APPROVED** [review]

File: core/pom.xml (hunk 2 above)
Comment (lukaszlenart) [reply core/pom.xml:247]: Do you mean to use `<scope>test</scope>`?

File: plugins/portlet-mocks/pom.xml (hunk 3 above)
Comment (lukaszlenart) [reply plugins/portlet-mocks/pom.xml:54]: It's the only way to avoid conflict of dependencies :D

File: core/pom.xml (hunk 2 above)
Comment (sepe81) [reply core/pom.xml:247]: I guess, for the test suite to work, this would suffice -- but my comment was just meant as a review note.

**Pull request merged**